internal/deb/extract.go

```diff
@@ -104,15 +105,14 @@ func Extract(pkgReader io.Reader, options *ExtractOptions) (err error) {
 
 func extractData(dataReader io.Reader, options *ExtractOptions) error {
 
+	oldUmask := syscall.Umask(0)
```

Comment:
Instead of touching umask, can we change the file-creation logic to make sure files are created with the proper permissions?

Reply:
I don't think we can. `open(2)` is the syscall we ultimately call, and that one respects umask. Alternatively, we can change `fsutil.Create` so that it calls `chmod(2)` after `open(2)` (`os.OpenFile`), but that's two syscalls, and it leaves file with incorrect permissions if we return on `chmod` failure without additional cleanup. So this is the simpler way.

Reply:
We already do the non-simple way in some cases (see fsutil/create.go), and this seems to be the best way as otherwise we'll have logic sprinkled through the code base which expects someone to call Umask outside of it, instead of doing what it was told (create the entry, with the given mode).  Also, the Umask change is global, which means that changing it breaks concurrent logic, and it also breaks the real meaning of the flags in case we need to create files taking the umask into account. For all those reasons, I would prefer to attempt the change suggested back in October, and actually enforce the mode when desired, despite the umask.

Reply:
@niemeyer All functions in `fsutil/create.go` are still affected by umask. The mode you pass to open/creat/mkdir system calls is AND-ed with umask. I don't know of any way to enforce the desired mode not affected by umask at the time of file/directory creation. The only way I know of is to create the file/directory and then `chmod` it. This means that for `os.Mkdir()` to actually create all directories with 0755, I will need to copy and rewrite `os.Mkdir()` in the chisel, to chmod each directory after it's created. Not saying that it's a problem, but are you OK with that?

The `syscall.Umask(0)` can be set just for the duration of the functions in `fsutil/create.go`. That can still break concurrent logic, but AFAIK we don't run any goroutines in chisel.

Reply:
@niemeyer For the record, we already set umask to 0 in at https://github.com/canonical/chisel/blob/main/internal/slicer/slicer.go#L53. Currently the checks in all PRs fail because GitHub Actions workers apparently run with umask `022` as opposed to default Ubuntu `002` on which the current code base is usually tested. Can we resume the discussion about this one?

Reply:
Thanks, I'll just merge it. I can change it later if I end up caring enough about this.